### PR TITLE
Add gcc-analyzer & clang-analyzer to CI

### PR DESCRIPTION
**Pull request by ojab**
Fixes https://github.com/ruby/yarp/issues/842, gcc added just because why not.
Both are failing, so need to be fixed to make them required checks

[Test run](https://github.com/ojab/yarp/actions/runs/5511118250)

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -99,3 +99,33 @@ jobs:
...
+    steps:
+    - uses: actions/checkout@v3
+    - name: Set up Ruby
+      uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: head
+        bundler-cache: true
+    - name: Run build with gcc-analyzer enabled
+      run: |
+        sed -e  's/CFLAGS :=$/CFLAGS := -fanalyzer/g' -i Makefile
```
Comment (ojab) [inline .github/workflows/main.yml:114]: This is fragile, but AFAIU currently it's not possible to pass CFLAGS to the `make` from the outside. Can create another target though.

**Comment by ojab** [comment]
>This workflow requires approval from a maintainer.

sending to review

**Comment by ojab** [comment]
Also coverity find _something_, cannot view what exactly because project is not approved for me [yet?], maybe makes sense for some maintainer to register & setup it (pretty simple, https://github.com/ojab/yarp/commit/a3539e5c564a941d3e14ef49545c067dad425acf)
image

**Comment by ojab** [comment]
One for ruby, rstring.h (should be filtered out)
image
and one for YARP, [`compile.c`](https://github.com/ruby/yarp/blob/main/ext/yarp/compile.c#L515-L527)
image

**Comment by kddnewton** [comment]
Hey @ojab this is great work! Sorry I haven't gotten a chance to dig in too much to this, but I just wanted to drop by and say I'm leaving it open because I intend to merge it at some point.

**Comment by ojab** [comment]
Changed gcc-analyzer to use `CFLAGS` since it's supported after switch to autotools.

**Comment by kddnewton** [comment]
@ojab I finally got around to fixing all these warnings. If you wouldn't mind, could you rebase off https://github.com/ruby/yarp/pull/1212? That should fix everything and then we can get these merged.

**Comment by ojab** [comment]
@kddnewton rebased

**Comment by kddnewton** [comment]
Thanks @ojab!

**Pull request merged**